cmd/root.go

```diff
@@ -199,6 +201,18 @@ func parseConfig(cmd *cobra.Command, conf *proxy.Config, args []string) error {
 		cmd.Println("Ignoring disable-traces as telemetry-project was not set")
 	}
 
+	if userHasSet("api-endpoint-url") && conf.ApiEndpointUrl != "" {
+		_, err := url.Parse(conf.ApiEndpointUrl)
+		if err != nil {
+			return newBadCommandError(fmt.Sprintf("the value provided for --api-endpoint is not a valid url, %v", conf.ApiEndpointUrl))
+		}
+
+		// add a trailing '/' if omitted
```

Comment:
I prefer this to the v1 behavior, which would exit  code 1 when the url did not have a trailing space.